inc/aiosp_common.php

```diff
@@ -182,4 +189,256 @@ public static function make_xml_safe( $tag, $value ) {
+			 *
+			 * @see aiosp_common::set_transient_url_postids()
+			 */
+			add_action( 'shutdown', array( 'aiosp_common', 'set_transient_url_postids' ) );
+		}
+
+		return $id;
+	}
+
+	/**
+	 * Sets the transient data at the last hook instead at every call.
+	 *
+	 * @see set_transient()
+	 * @link https://developer.wordpress.org/reference/functions/set_transient/
+	 *
+	 * @since 2.9.2
+	 */
+	public static function set_transient_url_postids() {
+		if ( is_multisite() ) {
+			set_site_transient( 'aioseop_multisite_attachment_url_postids', self::$attachment_url_postids, 24 * HOUR_IN_SECONDS );
+		} else {
+			set_transient( 'aioseop_attachment_url_postids', self::$attachment_url_postids, 24 * HOUR_IN_SECONDS );
+		}
+
+	}
+
+	/**
+	 * Attachment URL to Post ID - Query 1
+	 *
+	 * This is intended to work solely with `aiosp_common::attachment_url_to_post_id()`. Calling this multiple times
+	 * is memory intense.
+	 *
+	 * @see wpdb::get_results()
+	 * @link https://developer.wordpress.org/reference/classes/wpdb/get_results/
+	 *
+	 * @return array
+	 */
+	public static function attachment_url_to_postid_query_1() {
+		global $wpdb;
+
+		$results_1 = $wpdb->get_results(
+			"SELECT ID, MD5(guid) AS guid FROM $wpdb->posts WHERE post_type='attachment' AND post_status='inherit' AND post_mime_type LIKE 'image/%';",
+			ARRAY_A
+		);
+
+		if ( $results_1 ) {
+			$results_1 = array_combine(
+				wp_list_pluck( $results_1, 'guid' ),
+				wp_list_pluck( $results_1, 'ID' )
+			);
+		} else {
+			$results_1 = array();
+		}
+
+		return $results_1;
+	}
+
+	/**
+	 * Attachment URL to Post ID - Query 2
+	 *
+	 * Unused/Conceptual function. This is intended to work solely with `aiosp_common::attachment_url_to_post_id()`.
+	 * Calling this multiple times is memory intense. It's intended to query for custom images, and data for those types
+	 * of images only exists in the postmeta database table
+	 *
+	 * @todo Investigate unserialize() memory consumption/leak.
+	 * @link https://www.evonide.com/breaking-phps-garbage-collection-and-unserialize/
+	 *
+	 * @see aiosp_common::attachment_url_to_postid()
+	 * @see unserialize()
+	 * @link http://php.net/manual/en/function.unserialize.php
+	 * @see wpdb::get_results()
+	 * @link https://developer.wordpress.org/reference/classes/wpdb/get_results/
+	 * @see wp_upload_dir()
+	 * @link https://developer.wordpress.org/reference/functions/wp_upload_dir/
+	 *
+	 * @return array
+	 */
+	public static function attachment_url_to_postid_query_2() {
+		global $wpdb;
+
+		$tmp_arr = array();
+		$results_2 = $wpdb->get_results(
+			"SELECT post_id, meta_value FROM {$wpdb->postmeta} WHERE `meta_key` = '_wp_attachment_metadata' AND `meta_value` != '" . serialize( array() ) . "';",
+			ARRAY_A
+		);
+		if ( $results_2 ) {
+			for ( $i = 0; $i < count( $results_2 ); $i++ ) {
+				// TODO Investigate potentual memory leak(s); currently with unserialize.
+				$meta_value = unserialize( $results_2[ $i ]['meta_value'] );
```

Comment:
maybe https://developer.wordpress.org/reference/functions/maybe_unserialize/ is better?

Reply:
It should be expected as serialized values, but wouldn't hurt to add/change.

However, this is in a disabled method for conceptual use.